### PR TITLE
JZ - CommonStats

### LINKED ISSUE #35: FEATURE: CommonStats Entity/Repository (break from #13)
# User Story

* As an admin
* I can track numCows and weightedAvgCowHealth every 6 hours
*  so that I can better track what's happening over the course of the game

### Discussion

There is an existing feature to generate reports.

In addition, and in a similar manner, Prof. deVries would like to track a small number of statistics over time.  Here's what he says:

> What would be nice would be a simple report showing just the overall health and total > number of cows of a chosen commons as a function of time. The time interval could
> be 6 hours or something like that.
> Would that be hard?

### Implementation Todos

Backend for stats
- [ ] There is a CommonsStats entity with id, commonsId, numCows, avgHealth, and a timestamp (make the timestamp similar to the one for reports and reportlines), and a CommonsStatsRepository to go along with it.

### PR DESCRIPTION
**Pull request by jczhou02**
closes #35 

Making a new database table for CommonStats of all commons and attributes: id, commonsId, numCows, avgHealth, and a timestamp. 
I decided to go with a CrudRepository since that is what the other report repository instances use and the User Story does not need any higher level methods such as paging or sorting, etc.

### CODE REVIEW

**Review by kirrarista — APPROVED** [review]
Looks good! Will look at it more thoroughly when the front end is up

**Review by pconrad — APPROVED** [review]
LGTM

File: src/main/java/edu/ucsb/cs156/happiercows/entities/CommonStats.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+@Entity(name = "commonstats")
+
+public class CommonStats {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+    private long commonsId;
+    private int numCows;
+    private double avgHealth;
+    private LocalDateTime timestamp;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/entities/CommonStats.java:28]:
```suggestion
    @CreatedDate
    private LocalDateTime timestamp;
```

If we add this, with import:
```java
import org.springframework.data.annotation.CreatedDate;
```

then the field will be automatically populated with the correct value when you save a new record in the database.

**Review by pconrad — CHANGES_REQUESTED** [review]
see my comment

**Review by rockygao2020 — APPROVED** [review]
Looks good, tested Dokku after deleting databases

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**